src/common/src/util/value_encoding/column_aware_row_encoding.rs

```diff
@@ -23,97 +23,149 @@ use std::collections::HashSet;
 use std::sync::Arc;
 
 use ahash::HashMap;
-use bitflags::bitflags;
+use bitfield_struct::bitfield;
 
 use super::*;
 use crate::catalog::ColumnId;
 
-bitflags! {
-    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-    struct Flag: u8 {
-        const EMPTY = 0b_1000_0000;
-        const OFFSET8 = 0b01;
-        const OFFSET16 = 0b10;
-        const OFFSET32 = 0b11;
```

Comment:
The `OFFSET32` here overlaps with `OFFSET8` and `OFFSET16`, for which the bitflag doesn't seem to provide clear semantics. That's why I decide to re-implement with bitfield.